### DISCUSSION TITLE
Everyday Usage - QT - version limiting

### DISCUSSION THREAD

**Discussion by scottfurry** [Homebrew/discussions#4493] [Everyday usage] [answered]
### Output of `brew config`

```shell
% brew config 
HOMEBREW_VERSION: 4.0.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0f8068573bbf886fdc78d3f9daca54d53e7e49e
Last commit: 12 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5cb1fffc2c20e18e3e187e89e8c7fdc521d20cd8
Core tap last commit: 5 weeks ago
Core tap branch: master
Core tap JSON: 08 May 19:33 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.5-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  xalan-c
```

### Description of issue

I'm finding it difficult to find a solution to link against QT libraries. In researching the matter, dated posts mention problems with Qt4/Qt5 versions. Looking at installed kegs/casks, there are multiple dependencies to just `qt` which pulls in Qt6. I now have a messy situation of mixing of Qt5 and Qt6 on system.

So as to ease development/setup is there a mechanism to have only one version of Qt to be installed. Is there a some way to block or blacklist a Qt version in order to avoid the formulae calling for `qt` defaulting installation of `qt@6`?

**Comment by SMillerDev** [comment, marked as answer]
There is not. Unversioned formulae always point to the latest version so anything depending on qt will pull in 6.

**Reply by Bo98** [reply]
For a reason why: formulae would have to be completely recompiled if you swap Qt versions. They are not drop-in replacements for each other.

Qt5 and Qt6 is able to be installed side by side fine though.
